libbeat/common/jsontransform/jsonhelper.go

```diff
@@ -39,8 +39,8 @@ var (
 // WriteJSONKeys writes the json keys to the given event based on the overwriteKeys option and the addErrKey
 func WriteJSONKeys(event *beat.Event, keys map[string]interface{}, expandKeys, overwriteKeys, addErrKey bool) {
 	if expandKeys {
-		if err := expandFields(keys); err != nil {
-			event.SetErrorWithOption(createJSONError(err.Error()), addErrKey)
+		if err := expandFields(keys); err != nil && addErrKey {
```

Comment:
I understand what you tried to do here, and i agree it'll save a few CPU cycles, but in favor of consistency, i. would avoid making this optimization. Having this in place implies `WriteJSONKeys` knows the details of `SetErrorWithOption`.

If in the future there will be new content in `SetErrorWithOption` outside the `if` statement, we will have a bug.

Reply:
That was unintentional, sorry about that